sherpa/bin/streaming_pruned_transducer_statelessX/streaming_server.py

```diff
@@ -483,8 +507,23 @@ async def handle_connection_impl(
 
             while len(stream.features) > self.chunk_length:
                 await self.compute_and_decode(stream)
-                result = self.beam_search.get_texts(stream)
-                await socket.send(result)
+                hyp = self.beam_search.get_texts(stream)
+
+                if stream.endpoint_detected(self.online_endpoint_config):
```

Comment:
Maybe something like : 
```
is_final = stream.endpoint_detected(self.online_endpoint_config)
if is_final:
    self.beam_search.init_stream(stream)
    stream.segment += 1

message = {"segment": stream.segment - 1,
                     "text": hyp,
                    "final": is_final}
```

Reply:
Thanks! That sounds good. I just made the changes in the latest commit.